java/ql/src/semmle/code/java/dataflow/internal/ContainerFlow.qll

```diff
@@ -112,57 +112,63 @@ private predicate taintPreservingQualifierToMethod(Method m) {
   // java.util.Map
   m
       .(MapMethod)
-      .hasName(["compute", "computeIfAbsent", "computeIfPresent", "entrySet", "get", "getOrDefault",
-            "merge", "putIfAbsent", "remove", "replace", "values"])
+      .hasName(["computeIfAbsent", "entrySet", "get", "getOrDefault", "put", "putIfAbsent",
+            "remove", "replace", "values"])
   or
   // java.util.Collection
   m.(CollectionMethod).hasName(["parallelStream", "stream", "toArray"])
   or
   // java.util.List
   m.(CollectionMethod).hasName(["get", "listIterator", "set", "subList"])
   or
-  m.(CollectionMethod).hasName("remove") and
-  (m.getNumberOfParameters() = 0 or m.getParameterType(0).(PrimitiveType).hasName("int"))
+  m.(CollectionMethod).hasName("remove") and m.getParameterType(0).(PrimitiveType).hasName("int")
   or
   // java.util.Vector
   m.(CollectionMethod).hasName(["elementAt", "elements", "firstElement", "lastElement"])
   or
   // java.util.Stack
-  m.(CollectionMethod).hasName(["peek", "pop", "push"])
+  m.(CollectionMethod).hasName(["peek", "pop"])
   or
   // java.util.Queue
-  m.(CollectionMethod).hasName(["element", /*"peek", "remove"*/ "poll"])
```

Comment:
Did you remove the comment for `peek` by accident?
It would probably good to add a comment as well:
```
// covered by Stack: peek()
```